assets/src/block-editor/QueryBlockExtension/QueryBlockExtension.js

```diff
@@ -1,154 +1,125 @@
+                .getBlock(wp.data.select('core/block-editor')
+                .getBlockParents(selectedBlock.clientId)[0]);
+
+              if(parentBlock && parentBlock.attributes.namespace === 'planet4-blocks/actions-list') {
+                setPostTemplate(selectedBlock);
               }
-            }, 50);
-
-            return () => {
-              clearTimeout(timeout);
-            };
-          }
-        }, [props.isSelected]);
+            }
+          }, [selectedBlock]);
+
+          useEffect(() => {
+            // Reset every time a new block is selected
+            setPostTemplate(null)
+            setSelectedBlock(wp.data.select('core/block-editor').getSelectedBlock());
+          }, [
+            wp.data.select('core/block-editor').getSelectedBlock(),
+          ]);
+
+          return useMemo(() => (
+            <>
+              <InspectorControls>
+                <PanelBody
+                  title={__('Layout view', 'planet4-blocks-backend')}
+                  initialOpen={true}
+                >
+                  {attributes.layout && (
+                    <>
+                      <RadioControl
+                        label={__('Display as', 'planet4-blocks-backend')}
+                        help={__('The layout of the action|post list', 'planet4-blocks-backend')}
+                        selected={attributes.layout.type}
+                        options={layoutTypes}
+                        onChange={type => onChangeHandler(type)}
+                      />
+                    </>
+                  )}
+                </PanelBody>
+              </InspectorControls>
+              <BlockEdit {...props} />
+            </>
+          ), [
+            attributes,
+            isPostTemplate,
+            selectedBlock,
+            wp.data.select('core/block-editor').getSelectedBlock()
+          ]);
+        }
 
-        return useMemo(() => (
-          <>
-            {isSelectedBlock('planet4-blocks/actions-list') && (
-              <>
-                <InspectorControls>
-                  <PanelBody
-                    title={__('Layout view', 'planet4-blocks-backend')}
-                    initialOpen={true}
-                  >
-                    <RadioControl
-                      label={__('Display as', 'planet4-blocks-backend')}
-                      help={__('The layout of the action|post list', 'planet4-blocks-backend')}
-                      selected={attributes.displayLayout.type}
-                      options={[
-                        {label: 'Grid', value: 'flex'},
-                        {label: 'Carousel', value: 'carousel'},
-                      ]}
-                      onChange={value => {
-                        onChangeHandler(value);
-                      }}
-                    />
-                  </PanelBody>
-                </InspectorControls>
-                <BlockControls group="other">
```

Comment:
We don't this too. Why? Because we add the layout options only to the `core/query`